### ISSUE TITLE
Multiple argument paas

### ISSUE THREAD

**Issue by prasenforu** [adnanh/webhook#208]
hooks.json
```
[
   {
     "id": "createjob",
     "execute-command": "/etc/webhook/test.sh",
     "command-working-directory": "/etc/webhook/",
     "pass-arguments-to-command":
     [
      {
        "source": "url",
        "name": "usrname"
      },
      {
        "source": "url",
        "name": "giturl"
      },
      {
        "source": "url",
        "name": "cfnname"
      },
      {
        "source": "url",
        "name": "stkname"
      },
      {
        "source": "url",
        "name": "acckey"
      },
      {
        "source": "url",
        "name": "seckey"
      },
      {
        "source": "url",
        "name": "jobname"
      },
      {
        "source": "url",
        "name": "param"
      }
     ]
   }
]
```

my script as follows

```
#!/bin/sh

# This script job creation using webhook

usrname=$1
giturl=$2
cfnname=$3
stkname=$4
acckey=$5
seckey=$6
jobname=$7
param=$8

echo $usrname
echo $giturl
echo $cfnname
echo $stkname
echo $acckey
echo $seckey
echo $jobname
echo $param
```

CURL command executing -

curl http://ip-10-128-240-67:9000/hooks/createjob?usrname=kar002p&giturl=https://github.com/prasenforu/openshift-aws.git&cfnname=test.yml&stkname=awsec2&acckey=hdfhHFSFHS&seckey=PRASENJITkar&jobname=EC2-NEW&param=asd 
[1] 11550
[2] 11551
[3] 11552
[4] 11553
[5] 11554
[6] 11555
[7] 11556
[root@ip-10-128-240-67 webhook]#
[1]   Done                    curl http://ip-10-128-240-67:9000/hooks/createjob?usrname=kar002p
[2]   Done                    giturl=https://github.com/prasenforu/openshift-aws.git
[3]   Done                    cfnname=test.yml
[4]   Done                    stkname=awsec2
[5]   Done                    acckey=hdfhHFSFHS
[6]-  Done                    seckey=PRASENJITkar
[7]+  Done                    jobname=EC2-NEW

Logs output -

```
[webhook] 2017/12/06 15:19:48 createjob got matched
[webhook] 2017/12/06 15:19:48 createjob hook triggered successfully
[webhook] 2017/12/06 15:19:48 2017-12-06T15:19:48Z | 200 |       44.934Âµs | ip-10-128-240-67:9000 | GET /hooks/createjob
[webhook] 2017/12/06 15:19:48 error extracting command arguments: couldn't retrieve argument for {Source:url Name:giturl EnvName:}
[webhook] 2017/12/06 15:19:48 error extracting command arguments: couldn't retrieve argument for {Source:url Name:cfnname EnvName:}
[webhook] 2017/12/06 15:19:48 error extracting command arguments: couldn't retrieve argument for {Source:url Name:stkname EnvName:}
[webhook] 2017/12/06 15:19:48 error extracting command arguments: couldn't retrieve argument for {Source:url Name:acckey EnvName:}
[webhook] 2017/12/06 15:19:48 error extracting command arguments: couldn't retrieve argument for {Source:url Name:seckey EnvName:}
[webhook] 2017/12/06 15:19:48 error extracting command arguments: couldn't retrieve argument for {Source:url Name:jobname EnvName:}
[webhook] 2017/12/06 15:19:48 error extracting command arguments: couldn't retrieve argument for {Source:url Name:param EnvName:}
**[webhook] 2017/12/06 15:19:48 executing /etc/webhook/test.sh (/etc/webhook/test.sh) with arguments ["/etc/webhook/test.sh" "kar002p" "" "" "" "" "" "" ""] and environment [] using /etc/webhook/ as cwd
[webhook] 2017/12/06 15:19:48 command output:**

[webhook] 2017/12/06 15:19:48 finished handling createjob

```

Basically its taking 1st variable, rest are not taking.

**Comment by adnanh** [comment]
PEBKAC :-)

Here's what's happening:

You are trying to execute `curl http://ip-10-128-240-67:9000/hooks/createjob?usrname=kar002p&giturl=https://github.com/prasenforu/openshift-aws.git&cfnname=test.yml&stkname=awsec2&acckey=hdfhHFSFHS&seckey=PRASENJITkar&jobname=EC2-NEW&param=asd`

But, `&` is a special character in shell which is interpreted by sending the command before the `&` to the background.

So this command is split into several commands:
1 => `curl http://ip-10-128-240-67:9000/hooks/createjob?usrname=kar002p`
2 => `usrname=kar002p`
3 => ...

Those numbers you see in the console output after you run the command are PIDs from those sub-commands. And the "Done ..." messages are the result of your shell finishing the execution of each command. 

So what webhook is receiving is the `?usrname=kar002p`, and that's everything it can parse, rest of the arguments are never sent to the webhook :-)

In order to fix this, wrap the URL with quotes like so `curl "http://ip-10-128-240-67:9000/hooks/createjob?usrname=kar002p&giturl=https://github.com/prasenforu/openshift-aws.git&cfnname=test.yml&stkname=awsec2&acckey=hdfhHFSFHS&seckey=PRASENJITkar&jobname=EC2-NEW&param=asd"`

Hope this clears up some things. :-)

**Comment by prasenforu** [comment]
thanks...

**Comment by adnanh** [comment]
Good luck, if you have any other issues, don't hesitate to open up another issue :-)

**Comment by prasenforu** [comment]
Its continue with current issue ...

When executing below command 

`curl "http://ip-10-128-240-67:9000/hooks/createjob?usrname=kar002p&giturl=https://github.com/prasenforu/openshift-aws.git&cfnname=test.yml&stkname=awsec2&acckey=hdfhHFSFHS&seckey=/VhBo3twtjPnfvVRzppm+M1wkmWCp8UENTa6skkx&jobname=EC2-NEW&param=asd"`

if can see very closely there is parameter (seckey=/VhBo3twtjPnfvVRzppm+M1wkmWCp8UENTa6skkx) I am passing where plus sign (+).

In webhook log its coming as blank or white space.

```
[webhook] 2017/12/07 13:28:15 [71a6d5] createjob hook triggered successfully
[webhook] 2017/12/07 13:28:15 [71a6d5] executing /etc/webhook/test.sh (/etc/webhook/test.sh) with arguments ["/etc/webhook/test.sh" "kar002" "https://github.com/prasenforu/openshift-aws.git" "test.yml" "awsec2" "hdfhHFSFHS" "/VhBo3twtjPnfvVRzppm M1wkmWCp8UENTa6skkx" "EC2-NEW" "asd"] and environment [] using /etc/webhook/ as cwd
```

**Comment by adnanh** [comment]
You will have to [URL encode](https://www.w3schools.com/tags/ref_urlencode.asp) the parameters (stuff after each `=`) so that they are safe to be transmitted via HTTP.

**Comment by prasenforu** [comment]
can u give some example PLEASE.

**Comment by adnanh** [comment]
You will have to rewrite your curl command like so `curl -G "http://ip-10-128-240-67:9000/hooks/createjob" --data-urlencode "usrname=kar002p" --data-urlencode "giturl=https://github.com/prasenforu/openshift-aws.git"` etc ...

**Comment by prasenforu** [comment]
Perfect :) it works ... :+1: 

BTW in normal browser it should work with as follows .. right (no option to test in browser)

`http://ip-10-128-240-67:9000/hooks/createjob?usrname=kar002p&giturl=https://github.com/prasenforu/openshift-aws.git&cfnname=test.yml&stkname=awsec2&acckey=hdfhHFSFHS&seckey=/VhBo3twtjPnfvVRzppm+M1wkmWCp8UENTa6skkx&jobname=EC2-NEW&param=asd`

Right now I have no option to test in browser, that is why asking.
